site/src/comparison.rs

```diff
@@ -0,0 +1,583 @@
+//! Functionality for comparing
+//! This is mainly used to build the triage report and the perf
+//! comparison endpoints
+
+use crate::api;
+use crate::db::{self, ArtifactId, Cache, Crate, Profile};
+use crate::load::InputData;
+use crate::selector::{self, Tag};
+
+use collector::Bound;
+use database::Date;
+use serde::Serialize;
+
+use std::collections::HashMap;
+use std::error::Error;
+use std::sync::Arc;
+
+type BoxedError = Box<dyn Error + Send + Sync>;
+
+pub async fn handle_triage(
+    body: api::triage::Request,
+    data: &InputData,
+) -> Result<api::triage::Response, BoxedError> {
+    let start = body.start;
+    let end = body.end;
+    // Compare against self to get next
+    let master_commits = rustc_artifacts::master_commits().await?;
```

Comment:
Looked into this more. Indeed this would save us having to do the first `compare` to get the next commit, but we still need to fetch the `master_commits` in order to get the PR number. I tried doing a db lookup for each PR number but this didn't seem to work. 

Either way, perhaps this should be a follow up to this PR since there doesn't seem to be a trivial way to improve this.

Reply:
Ah, that makes sense. Yeah, we can leave fine tuning to later; we do store the PR number I think but it'll be fine to do that as a follow-up.